### ISSUE TITLE
User Registration Integration password isn't set correctly

### ISSUE THREAD

**Issue by cookie10codes** [verbb/formie#1962]
### Describe the bug

When I create my form and map the fields for the registration, the registration succeeds but I can't login after verifying. I need to reset met password in order to get it to work.

I think there's something mismachting according to passwords?

Tested it out via https://verbb.io/craft-plugins/formie/user-guides/create-a-content-managed-user-registration-form-with-formie and this one works. But I want the default form as the other ones, so the integration would be the best option.

### Steps to reproduce

1. Create a form
2. Enable User Registration
3. Map the fields accordingly

<img width="1610" alt="Screenshot 2024-06-28 at 19 22 02" src="https://github.com/verbb/formie/assets/20947573/84c92705-9b02-489f-8597-144d8811f32d">

### Form settings

- Multi-page form: No
- Submission Method: Page Reload
- Client-side Validation: No
- Custom Form Templates: No

### Craft CMS version

Craft Pro 5.2.3

### Plugin version

3.0.0-beta.16

### Multi-site?

no

**Comment by engram-design** [comment]
Can you comment on what sort of field you're mapping to the Password attribute? 

Using a Password field on my end shows the results as expected. The password will be hashed and stored in the `users` database table, and when I receive my activation email (as per your settings), and activate my account, I can login with the same password that I used filling out the form.

**Comment by cookie10codes** [comment]
So I have two password field, the first one with the field handle "password" and the second one with "newPassword". Within the User Registration Integration, I attach the newPassword to the password field.

I also tried the password field, both didn't work. My user gets activated after clicking the verification code.

Maybe the issues lies somewhere else or I'm doing something wrong with the password field.

It only works when I'm doing a forget password process, then I'm able to login

**Comment by engram-design** [comment]
Are both fields a Password Formie field type though? It sounds like it is, I just wanted to confirm. The handles can be whatever you like, as mapping takes care of things.

I'll keep at trying to replicate this one...

**Comment by cookie10codes** [comment]
@engram-design affirmative, they both are

**Comment by amici-infotech** [comment]
@engram-design I am facing the same issue in Craft cms 5.2.4.1 with Formie 3.0.0-beta.16

User fill the registration form and gets activation email, which user clicks and logged in. But, after logout, they cant login again with the same password.

I checked the db table and found the Password is hashed and stored as well. I dont know what happens in middle that broke it.

Here are ss of my mapping fields
![image](https://github.com/verbb/formie/assets/20450552/f6868c69-98a9-45a0-a757-593f38efabfc)
![image](https://github.com/verbb/formie/assets/20450552/34cf5a60-aa9f-4645-bc4b-5eb3f67cae65)

**Comment by amici-infotech** [comment]
@engram-design  I think I found it. In `vendor/verbb/formie/src/integrations/elements/User.php` Line Number 385 has this code.

```
$passwordFieldHandle = str_replace(['{', '}'], ['', ''], $passwordFieldHandle);
```

which I replaced with this:
```
$passwordFieldHandle = str_replace(['{field:', '}'], ['', ''], $passwordFieldHandle);
```

and now it works.

**Comment by cookie10codes** [comment]
@amici-infotech nice find!

**Comment by engram-design** [comment]
I have no idea why that's not causing an error on my end, but that's correct - that'd be the problem! Love your work.

Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-5 as 3.0.0-beta.16"`.

**Comment by engram-design** [comment]
Fixed in [3.0.0-beta.17](https://github.com/verbb/formie/releases/tag/3.0.0-beta.17)
